--- a/src/node_sea.cc
+++ b/src/node_sea.cc
@@ -401,29 +401,35 @@ std::optional<SeaConfig> ParseSingleExecutableConfig(
       return std::nullopt;
     }
     if (key == "main") {
-      if (field.value().get_string().get(result.main_path) ||
-          result.main_path.empty()) {
+      std::string_view main_path_view = result.main_path;
+      if (field.value().get_string().get(main_path_view) ||
+          main_path_view.empty()) {
         FPrintF(stderr,
                 "\"main\" field of %s is not a non-empty string\n",
                 config_path);
         return std::nullopt;
       }
+      result.main_path = main_path_view;
     } else if (key == "output") {
-      if (field.value().get_string().get(result.output_path) ||
-          result.output_path.empty()) {
+      std::string_view output_path_view = result.output_path;
+      if (field.value().get_string().get(output_path_view) ||
+          output_path_view.empty()) {
         FPrintF(stderr,
                 "\"output\" field of %s is not a non-empty string\n",
                 config_path);
         return std::nullopt;
       }
+      result.output_path = output_path_view;
     } else if (key == "executable") {
-      if (field.value().get_string().get(result.executable_path) ||
-          result.executable_path.empty()) {
+      std::string_view executable_path_view = result.executable_path;
+      if (field.value().get_string().get(executable_path_view) ||
+          executable_path_view.empty()) {
         FPrintF(stderr,
                 "\"executable\" field of %s is not a non-empty string\n",
                 config_path);
         return std::nullopt;
       }
+      result.output_path = executable_path_view;
     } else if (key == "disableExperimentalSEAWarning") {
       bool disable_experimental_sea_warning;
       if (field.value().get_bool().get(disable_experimental_sea_warning)) {
