diff --git a/community/php85/APKBUILD b/community/php85/APKBUILD
index 2bf382333d3..bb5dd5032e3 100644
--- a/community/php85/APKBUILD
+++ b/community/php85/APKBUILD
@@ -25,20 +25,21 @@
 # json      | https://wiki.php.net/rfc/always_enable_json
 
 maintainer="Andy Postnikov <apostnikov@gmail.com>"
-pkgname=php85
+_pkgname=php85
+pkgname=$_pkgname-zts
 pkgver=8.5.8
 pkgrel=0
 _apiver=20250925
-_suffix=${pkgname#php}
+_suffix=${_pkgname#php}
 # Is this package the default (latest) PHP version?
 _default_php="yes"
-provides="$pkgname-cli php-cli php"  # for backward compatibility
+provides="$_pkgname-cli php-cli php"  # for backward compatibility
 provider_priority=100
 pkgdesc="The PHP$_suffix language runtime engine"
 url="https://www.php.net/"
 arch="all"
 license="PHP-3.01 BSD-3-Clause LGPL-2.0-or-later MIT Zend-2.0 Apache-2.0"
-depends="$pkgname-common"
+depends="$pkgname-common !$_pkgname"
 depends_dev="$pkgname=$pkgver-r$pkgrel autoconf pcre2-dev re2c"
 # Most dependencies between extensions is auto-discovered (see _extension()).
 _depends_mysqlnd="$pkgname-openssl"
@@ -98,9 +99,9 @@ subpackages="$pkgname-dbg $pkgname-dev $pkgname-doc
 #source="https://downloads.php.net/~daniels/php-${pkgver}RC1.tar.xz
 # stable releases
 source="https://www.php.net/distributions/php-$pkgver.tar.xz
-	$pkgname-fpm.initd
-	$pkgname-fpm.logrotate
-	$pkgname-module.conf
+	$_pkgname-fpm.initd
+	$_pkgname-fpm.logrotate
+	$_pkgname-module.conf
 	disabled-tests.list
 	disabled-tests.loongarch64.list
 	disabled-tests.x86.list
@@ -109,14 +110,14 @@ source="https://www.php.net/distributions/php-$pkgver.tar.xz
 	install-pear.patch
 	includedir.patch
 	sharedir.patch
-	$pkgname-fpm-version-suffix.patch
+	$_pkgname-fpm-version-suffix.patch
 	phpinfo-avif.patch
 	"
 builddir="$srcdir/php-$pkgver"
 
-_libdir="/usr/lib/$pkgname"
+_libdir="/usr/lib/$_pkgname"
 _extension_dir="$_libdir/modules"
-_extension_confd="/etc/$pkgname/conf.d"
+_extension_confd="/etc/$_pkgname/conf.d"
 
 _extensions="
 	bcmath
@@ -320,12 +321,12 @@ _build() {
 		--prefix=/usr \
 		--program-suffix=$_suffix \
 		--libdir=$_libdir \
-		--datadir=/usr/share/$pkgname \
-		--sysconfdir=/etc/$pkgname \
+		--datadir=/usr/share/$_pkgname \
+		--sysconfdir=/etc/$_pkgname \
 		--localstatedir=/var \
 		--with-layout=GNU \
 		--with-pic \
-		--with-config-file-path=/etc/$pkgname \
+		--with-config-file-path=/etc/$_pkgname \
 		--with-config-file-scan-dir=$_extension_confd \
 		--disable-rpath \
 		--disable-short-tags \
@@ -343,21 +344,24 @@ build() {
 	# rebuilds them once after `make distclean` clears the autoconf cache.
 	_build --disable-all --disable-phpdbg --disable-cli --with-apxs2
 	# Stash apache2 module outside $builddir; distclean will wipe sapi/.
-	install -Dm755 libs/libphp.so "$srcdir"/mod_php$_suffix.so
+	install -Dm755 libs/libphp.so /tmp/mod_php$_suffix.so
 	make distclean
 
 	local enable_litespeed
 	[ -z ${subpackages##*-litespeed*} ] && enable_litespeed=--enable-litespeed
 	_build $_shared_extensions \
 		--enable-phpdbg \
-		--with-pear=/usr/share/$pkgname \
+		--with-pear=/usr/share/$_pkgname \
 		--enable-fpm \
 		--with-fpm-acl \
 		$enable_litespeed \
-		--enable-embed
+		--enable-embed \
+		--with-readline \
+		--enable-zts \
+		--disable-zend-signals
 
 	# Restore apache2 module to where package() expects it.
-	install -Dm755 "$srcdir"/mod_php$_suffix.so \
+	install -Dm755 /tmp/mod_php$_suffix.so \
 		sapi/apache2handler/mod_php$_suffix.so
 }
 
@@ -392,7 +396,7 @@ package() {
 	# zend_test is needed at check() time but not shipped to end users.
 	rm "$pkgdir"/usr/lib/php$_suffix/modules/zend_test.so
 
-	install -Dm644 php.ini-production "$pkgdir"/etc/$pkgname/php.ini
+	install -Dm644 php.ini-production "$pkgdir"/etc/$_pkgname/php.ini
 
 	local file; for file in pear peardev pecl; do
 		sed -i -e "s|/usr/bin/php|/usr/bin/php$_suffix|g" \
@@ -415,6 +419,7 @@ package() {
 }
 
 dev() {
+	depends="!$_pkgname-dev"
 	default_dev
 	replaces="php-dev"
 
@@ -430,19 +435,20 @@ dev() {
 }
 
 doc() {
+	depends="!$_pkgname-doc"
 	default_doc
 
 	cd "$builddir"
 
-	mkdir -p "$subpkgdir"/usr/share/doc/$pkgname
+	mkdir -p "$subpkgdir"/usr/share/doc/$_pkgname
 	cp CODING_STANDARDS.md EXTENSIONS LICENSE NEWS \
 		README* UPGRADING* \
-		"$subpkgdir"/usr/share/doc/$pkgname/
+		"$subpkgdir"/usr/share/doc/$_pkgname/
 }
 
 apache2() {
 	pkgdesc="PHP$_suffix Module for Apache2"
-	depends="$depends apache2"
+	depends="$depends apache2 !$_pkgname-apache2"
 	provides="php-apache2"
 
 	amove usr/lib/apache2/mod_php$_suffix.so
@@ -452,6 +458,7 @@ apache2() {
 
 phpdbg() {
 	pkgdesc="Interactive PHP$_suffix debugger"
+	depends="!$_pkgname-phpdbg"
 	provides="php-phpdbg"
 
 	amove usr/bin/phpdbg$_suffix
@@ -463,6 +470,7 @@ phpdbg() {
 
 embed() {
 	pkgdesc="PHP$_suffix Embedded Library"
+	depends="!$_pkgname-embed"
 	provides="php-embed"
 
 	amove usr/lib/php$_suffix/libphp.so
@@ -470,6 +478,7 @@ embed() {
 
 litespeed() {
 	pkgdesc="PHP$_suffix LiteSpeed SAPI"
+	depends="!$_pkgname-litespeed"
 	provides="php-lightspeed"
 
 	amove usr/bin/lsphp$_suffix
@@ -481,6 +490,7 @@ litespeed() {
 
 cgi() {
 	pkgdesc="PHP$_suffix Common Gateway Interface"
+	depends="!$_pkgname-cgi"
 	provides="php-cgi"
 
 	amove usr/bin/php-cgi$_suffix
@@ -493,27 +503,28 @@ cgi() {
 fpm() {
 	pkgdesc="PHP$_suffix FastCGI Process Manager"
 	provides="php-fpm"
+	depends="!$_pkgname-fpm"
 
-	amove etc/$pkgname/php-fpm* usr/sbin var
-	amove usr/share/$pkgname/fpm
+	amove etc/$_pkgname/php-fpm* usr/sbin var
+	amove usr/share/$_pkgname/fpm
 
 	local file; for file in php-fpm.conf php-fpm.d/www.conf; do
-		mv "$subpkgdir"/etc/$pkgname/$file.default \
-			"$subpkgdir"/etc/$pkgname/$file
+		mv "$subpkgdir"/etc/$_pkgname/$file.default \
+			"$subpkgdir"/etc/$_pkgname/$file
 	done
 
-	install -D -m 755 "$srcdir"/$pkgname-fpm.initd \
+	install -D -m 755 "$srcdir"/$_pkgname-fpm.initd \
 		"$subpkgdir"/etc/init.d/php-fpm$_suffix
 
-	install -D -m 644 "$srcdir"/$pkgname-fpm.logrotate \
+	install -D -m 644 "$srcdir"/$_pkgname-fpm.logrotate \
 		"$subpkgdir"/etc/logrotate.d/php-fpm$_suffix
 
-	mkdir -p "$subpkgdir"/var/log/$pkgname
+	mkdir -p "$subpkgdir"/var/log/$_pkgname
 }
 
 pear() {
 	pkgdesc="PHP$_suffix Extension and Application Repository"
-	depends="$pkgname $pkgname-xml"
+	depends="$pkgname $pkgname-xml !$_pkgname-xml"
 	provides="php-pear"
 
 	mkdir -p "$subpkgdir"/usr/bin
@@ -524,13 +535,13 @@ pear() {
 		fi
 	done
 
-	amove etc/$pkgname/pear.conf usr/share
+	amove etc/$_pkgname/pear.conf usr/share
 }
 
 common() {
 	pkgdesc="$pkgdesc (common config)"
 	provides="php-common $pkgname-zlib php-zlib $pkgname-json php-json"  # for backward compatibility
-	depends=""
+	depends="!$_pkgname-common"
 
 	amove etc
 	mkdir -p "$subpkgdir"/$_extension_confd "$subpkgdir"/$_extension_dir
@@ -538,6 +549,7 @@ common() {
 
 phar() {
 	_extension
+	depends="!$_pkgname-phar"
 
 	amove usr/bin/phar$_suffix.phar usr/bin/phar$_suffix
 
@@ -555,7 +567,7 @@ _extension() {
 	provides="php-$extname"
 
 	: "${extdepends:=$(_resolve_extension_deps "$extname")}"
-	depends="$depends $extdepends"
+	depends="$depends $extdepends !$_pkgname-${extname}"
 
 	local load_order=$(_extension_load_order "$extname")
 
@@ -610,19 +622,5 @@ _extensions_by_load_order() {
 	printf '%s\n' $list | sort -t ';' -k 1 | sed -E 's/\d+;//'
 }
 
-sha512sums="
-9c58a121f91a23ec78834cb89b504031ba4d5d8ecb501c3de2172a315f2dad8a1dc39234e39f7b0e70c278c880a13f1eb198858fa7f1ee028701b583dbfbde28  php-8.5.8.tar.xz
-7e08969f99a3ccaa0b6a9e5b91ce7dc8775978f096cc4eeffe378988fd52b61c78fc230b8e686521a458dabc1741bbf160039fa334cf5548281b5dec9820e1f6  php85-fpm.initd
-be2926ba61d0a7ef99178eb1c02e9e5a6698e9423f51b090f997e6ec056dbf8f87c2cd627dbc96eae86b1442c9c1b2df16ed406295023a29c02470ba284e4c69  php85-fpm.logrotate
-57ef049fd8e659eed6b1e5895b8c8c04356873b80fdc7d2c66a7e750efdb27e8a3cd6c5bfa35ceae75d8f0885f1915611619260154f0ef2e6c60a27b7cc7909e  php85-module.conf
-585508cdc86490fa85cb9d5a77522fcbc72b114bd79f47ae5b2b86c82bd6654570c8f480e26855b9fb669da9b2cde013e7fe1d664664b1f97dc704ad36509ba5  disabled-tests.list
-a8b68c261e69f8feafa1bd8d9d2bd64c405884db716b30b7fa7f1d285fd938a5400bd57444148541040bb0d7410b21271d737303973af085458f3b4f8ce342aa  disabled-tests.loongarch64.list
-6d8f18822eeb61e1ef2d24551dde2dadaf7607f81364392351e6400da63e4d6fd9ea5f433c5ab7e4de78191d2bcebe3172d4a3024f0ee21b2e1e2c3cf4ad73cc  disabled-tests.x86.list
-579444b30b3229535dfa68e9341ef096140cd26f1c74f0f920cffe23bac51e144c21a49bfe36fb7356066136ebf0baff6f7e37b8c384c051b430344b66ef5e10  disabled-tests.ppc64le.list
-4934def86c4d3817080ba152bc07d58ac6987f3070464e21c8e33e506184e760963cbf34db3aa66dbaceac1b2bf4daa49e60194b9b36290a946559a82e4590d6  disabled-tests.s390x.list
-d7bf35f424d8383e4bf6372d3ba8fca2ba1d1c435d1aea495a39fe79a5e17b31f96d373a33d777830412477be35a58be29eb852d7c6c012204e6004c4366fac2  install-pear.patch
-b4f802ec3e69190cfa6f2b00bfc9c7741e8300aba479728a00345cbde1c4f20b9ce46b08b861f8d4bb3c6fd6abf1e0f648783937ebef1fdc119fa93d2837a385  includedir.patch
-61b007983c56dabbd1841255b209c0b9f2ddb83c8ade0dbd97a0ef59157c90bc785fe2b9f11e43e9d0db540fd96802dbca4adaeea72da8780f2a1a2d9895c37f  sharedir.patch
-73ae19d2b02873f560c4a25ed79a7436857fc405298a0b1e271638bf5dcc00f0bca403085517a59ab65739f03a12e77a838a22f90800ce8a8550c60ca6e7cf84  php85-fpm-version-suffix.patch
-6eb4d3272b7030611c0654133d380c01d1cae9faae52e8b5972df2f867d02a7d0b023699cb838594908c25601856ad69adf086fe3927c4d81a36861cdd56fce2  phpinfo-avif.patch
-"
+sha512sums=""
+
