--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ LDFLAGS+=-L.
 PACKAGE=newsboat
 ## Do not include libnewsboat-ffi into the test executable, as it doesn't link
 ## and can't contain any tests anyway.
-CARGO_TEST_FLAGS=--workspace --exclude libnewsboat-ffi
+CARGO_TEST_FLAGS=--frozen --workspace --exclude libnewsboat-ffi
 
 ifeq (, $(filter $(MAKECMDGOALS),distclean run-i18nspector fmt))
 include config.mk
@@ -95,7 +95,7 @@ ifeq ($(PROFILE),1)
 BUILD_TYPE=debug
 else
 BUILD_TYPE=release
-CARGO_BUILD_FLAGS+=--release
+CARGO_BUILD_FLAGS+=--release --frozen
 endif
 
 ifdef CARGO_BUILD_TARGET
@@ -120,7 +120,7 @@ ASCIIDOCTOR=asciidoctor
 MSGFMT=msgfmt
 RANLIB?=ranlib
 AR?=ar
-CARGO=cargo
+CARGO=cargo auditable
 CLANG_TIDY?=clang-tidy
 
 TEXTCONV=./txt2h
